include/micm/solver/state.hpp

```diff
@@ -78,4 +89,48 @@ namespace micm
     for (auto& name : parameters.state_variable_names_)
       variable_map_[name] = index++;
   }
+
+  template<template<class> class MatrixPolicy>
+  inline void State<MatrixPolicy>::SetConcentrations(
+      const System& system,
+      const std::unordered_map<std::string, double>& species_to_concentration)
```

Comment:
The species concentrations will be for a set of grid cells, so might be something like:
```suggestion
      const std::unordered_map<std::string, std::vector<double>>& species_to_concentration)
```
We'll want to try to come up with an efficient-as-possible solution eventually, but for now I think it's fine to just get something that works and can be optimized later.

Reply:
Okay that makes sense. I addressed this in the latest commit.